lib/livebook_web/live/home_live/session_list_component.ex

```diff
@@ -258,11 +267,43 @@ defmodule LivebookWeb.HomeLive.SessionListComponent do
     {:noreply, assign(socket, sessions: sessions, order_by: order_by)}
   end
 
+  def handle_event("fork_session", %{"id" => session_id}, socket) do
+    session = Enum.find(socket.assigns.sessions, &(&1.id == session_id))
+    %{images_dir: images_dir} = session
+    data = Session.get_data(session.pid)
+    notebook = Notebook.forked(data.notebook)
+
+    origin =
+      if data.file do
+        {:file, data.file}
+      else
+        data.origin
+      end
+
+    {:noreply,
+     create_session(socket,
+       notebook: notebook,
+       copy_images_from: images_dir,
+       origin: origin
+     )}
+  end
+
+  def handle_event("disconnect_runtime", %{"id" => session_id}, socket) do
+    session = Enum.find(socket.assigns.sessions, &(&1.id == session_id))
+    Session.disconnect_runtime(session.pid)
+    refresh_memory_info()
+    {:noreply, socket}
+  end
+
   def format_creation_date(created_at) do
     time_words = created_at |> DateTime.to_naive() |> Livebook.Utils.Time.time_ago_in_words()
     time_words <> " ago"
   end
 
+  def refresh_memory_info do
+    send_update(__MODULE__, memory_timestamp: System.os_time(), id: "session-list")
```

Comment:
Any advantage of using the timestamp instead of sending memory measurement directly?

Reply:
Also, to show the memory error we now take the measurement in `HomeLive.mount`, maybe we should keep it updated there and pass downstream to the components?